### PR TITLE
Completed Week 3 assessment

### PR DESCRIPTION
**Pull request by JLODI**
Completed Week 3 Assessment

### CODE REVIEW

File: react-debugging.html [hunk 1]
```diff
@@ -63,11 +70,12 @@
           // No need to change anything ABOVE this line ^
 
           return(
+            <>
```
Comment (SunkissedQueen) [inline react-debugging.html:73]: Wonderful! Just a few changes lead by the console and now there is a game!

File: interview-questions.md [hunk 2]
```diff
@@ -10,58 +10,71 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 6. STRETCH: What is an anonymous function in JavaScript?
 
-  Your answer:
+  Your answer: an anonymous function is a function with a name, which is defined by function () {} After it's creation it is not accessible, so it's usually assigned to a variable, as a function as a value.
 
   Researched answer:
-
+  An anonymous function is a function declared without a name. Normally the function's name is declared after "function" or before function syntax. Thus, it can't be accessed unless it's assigned in a variable as a value. Anonymous functions are usually used as arguments passed to higher order functions.
 
 ## Looking Ahead: Terms for Next Week
```
Comment (SunkissedQueen) [inline interview-questions.md:65]: Such thorough definitions!

File: interview-questions.md [hunk 3]
```diff
@@ -10,58 +10,71 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 6. STRETCH: What is an anonymous function in JavaScript?
 
-  Your answer:
+  Your answer: an anonymous function is a function with a name, which is defined by function () {} After it's creation it is not accessible, so it's usually assigned to a variable, as a function as a value.
 
   Researched answer:
-
+  An anonymous function is a function declared without a name. Normally the function's name is declared after "function" or before function syntax. Thus, it can't be accessed unless it's assigned in a variable as a value. Anonymous functions are usually used as arguments passed to higher order functions.
```
Comment (SunkissedQueen) [inline interview-questions.md:62]: Good research!

File: interview-questions.md [hunk 4]
```diff
@@ -10,58 +10,71 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
   Your answer:
+  YARN stands for Yet Another React Negotiator. It's a package manager that creates the file structure for a React app. You can setup a new React app via the terminal with yarn commands. When you run yarn, it will install the base file structure and dependencies needed for your React app. When modifying the code of the React application, we make a components folder under src where we create and code new components and import them within the App.js of the app's root folder.
 
   Researched answer:
-
+Yarn is a packaging system developed for node.js. Yarn is used to create new React applications and will build the file structure for any new app via terminal. Using yarn to run the React application is similar to running a server, and the application can be stopped by hitting ctrl + c in the terminal. yarn create react-app "name-of-app" creates the director and installs the files and dependencies needed for the app. The developer then makes a components folder under src and also modifies App.js when developing the React app.
```
Comment (SunkissedQueen) [inline interview-questions.md:54]: Lots of great content. Yarn definitely saves us time by bringing all the necessary code snippets to get our coding projects to properly function.

File: interview-questions.md [hunk 5]
```diff
@@ -10,58 +10,71 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
   Researched answer:
+  JSX is a markup that allows the developer to write HTML into javascript and allow easy placement in the DOM. It is not required but highly recommended to use JSX as it's a helpful visual aid when modifying the UI inside javascript code. One big difference is that you would need to encapsulate all HTML elements within a div if you want all elements to be rendered properly in the JSX
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
   Your answer:
+  YARN stands for Yet Another React Negotiator. It's a package manager that creates the file structure for a React app. You can setup a new React app via the terminal with yarn commands. When you run yarn, it will install the base file structure and dependencies needed for your React app. When modifying the code of the React application, we make a components folder under src where we create and code new components and import them within the App.js of the app's root folder.
```
Comment (SunkissedQueen) [inline interview-questions.md:51]: Great acronym!

File: interview-questions.md [hunk 6]
```diff
@@ -10,58 +10,71 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 4. What is JSX? What is one notable difference between HTML and JSX?
 
   Your answer:
+  JSX is like the javascript version of html. It's a markup language that pre processes HTML (actually XML) into plain javascript before the file is sent to the browser. The render method in react uses JSX and it's encapsulated by "fragments" which are empty tags that look like this <></>. One big difference between HTML and JSX is that the JSX in React only returns one element. For example if you put a header and a paragraph tag by itself in the JSX, it will only render the first thing, the header. In order to render all of that, you will need to encapsulate all of those elements into a div.
 
   Researched answer:
+  JSX is a markup that allows the developer to write HTML into javascript and allow easy placement in the DOM. It is not required but highly recommended to use JSX as it's a helpful visual aid when modifying the UI inside javascript code. One big difference is that you would need to encapsulate all HTML elements within a div if you want all elements to be rendered properly in the JSX
```
Comment (SunkissedQueen) [inline interview-questions.md:44]: Yes, need that fragment container to deliver our words to the screen.

File: interview-questions.md [hunk 7]
```diff
@@ -10,58 +10,71 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+  React is a javascript based user interface development library and is widely used among web developers. It allows easy creation of dynamic applications with pre-built code (less coding). React's Virtual DOM (or document object model) is a highly performing and efficient way to render data by comparing its components previous states and updates items that the DOM that were changed, leaving the other components untouched.
 
 
 3. Which lifecycle method is required in a React class component?
 
   Your answer:
+  The constructor method is required when creating data that belongs to a class. It runs automatically when a class is instantiated. For example when you're setting up a component and instantiating the class, you would type "constructor(){} within the class's code block.
 
   Researched answer:
+During the "Mounting" component lifecycle, the constructor method is required and called before anything else in the component. It's called with "props", as arguments and should always call super(props) before anything else which will initiate the parent class's constructor method and allows the component to inherit methods from the parent class.
```
Comment (SunkissedQueen) [inline interview-questions.md:34]: That is the proper way to make a space to hold data (state object). However, not all class components store data. Some components only display information and will not require any storage. Yet all components will need a render method.

File: interview-questions.md [hunk 8]
```diff
@@ -10,58 +10,71 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 2. What is React? Why would you use it?
 
   Your answer:
+  React is an open source javascript library used to build user interfaces. It is widely used in the industry and maintained by Meta (originally Facebook). React is very modular and based on components which are separate reusable pieces of code. It saves a lot of time for building the front-end is very good for making web applications with dynamic data.
 
   Researched answer:
-
+  React is a javascript based user interface development library and is widely used among web developers. It allows easy creation of dynamic applications with pre-built code (less coding). React's Virtual DOM (or document object model) is a highly performing and efficient way to render data by comparing its components previous states and updates items that the DOM that were changed, leaving the other components untouched.
```
Comment (SunkissedQueen) [inline interview-questions.md:25]: Open sources, free sources, got to love this development community

File: interview-questions.md [hunk 9]
```diff
@@ -10,58 +10,71 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. What is `this` in JavaScript?
 
   Your answer:
+"This" is a javascript keyword that refers an object. It can be used with .notation method to access a value that's within in object. The method, or the function within an object can refer to its values using the key and "this", like "this.(name of key)"
 
   Researched answer:
```
Comment (SunkissedQueen) [inline interview-questions.md:15]: Splendid!

File: interview-questions.md [hunk 10]
```diff
@@ -10,58 +10,71 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. What is `this` in JavaScript?
```
Comment (SunkissedQueen) [inline interview-questions.md:10]: For your interview questions, please add more of your experience with the answers. Tell the story how your challenge questions helped you understand the concept or how you explained it to your pair partner. These interview questions are opportunities to practice sharing your thoughts with an interviewer/future clients.

File: code-challenges.test.js [hunk 11]
```diff
@@ -21,9 +21,36 @@
...
+// return array of fibonacci numbers
+
+let array = [1,1]
+const fibonacciMaker = (number) =>   {
+   for (let i=2; i<number; i++){
+     array[i] = array[i-2] + array[i-1]
+   }
+   return array
+}
+
```
Comment (SunkissedQueen) [inline code-challenges.test.js:47]: This is wild that the test still runs below the function. Best practices are not only for flow but to help another developer understand our code. In the test suite the flow is to develop the test first then follow it with out desired function. Other than that great approach. The code is clean and logical.

File: code-challenges.test.js [hunk 12]
```diff
@@ -35,9 +62,35 @@ const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
 const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
 // Expected output: [-823, 7, 23]
 
+// describe("oddsOnly", () => {
+//   it("returns an array of only odd numbers from least to greatest", () => {
+//     expect(oddsOnly("array of numbers")).toEqual("array of only odd numbers in order")
 
 // b) Create the function that makes the test pass.
 
+// Pseudo Code:
```
Comment (SunkissedQueen) [inline code-challenges.test.js:71]: Great planning

File: code-challenges.test.js [hunk 13]
```diff
@@ -35,9 +62,35 @@ const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
...
+// if number, check if odd with modulo
+// if odd, include in new array
+// if even, omit from new Array
+// return sorted new array from least to greatest
+
+const oddsOnly = (array) => {
+  let newArray = array.filter(
+    value => value % 2 !== 0 && typeof(value) === 'number'
+  )
+  return newArray.sort(((a,b)=>a-b))
```
Comment (SunkissedQueen) [inline code-challenges.test.js:85]: Looks like there is one extra set of parenthesis that came along for the ride

File: code-challenges.test.js [hunk 14]
```diff
@@ -52,5 +105,54 @@ const numbersToAdd2 = [0, 7, -8, 12]
...
+// Pseudo Code:
+// declare a function arraySum
+// set the parameter to an array
+// itterate through array
+// declare empty array newArray
+// return value of index 0 as is with conditional
+// from index 1 onwards, add previous value of newArray and move onto next
+// return newArray of values each containing accumulating sum
+
+const arraySum = (array) => {
```
Comment (SunkissedQueen) [inline code-challenges.test.js:126]: mission accomplished! Let's see if hof's can do some heavy lifting for you.

**Review by SunkissedQueen — APPROVED** [review]
Jennice, I appreciate what you have done in this assessment. You pour some hard work onto this and it shows in your indentation, your logic, your pseudocode, your answers. Great work! I encourage you to start using hof's and sharing more of your personal experience with each element on the interview questions.

File: code-challenges.test.js (hunk 13 above)
Comment (JLODI) [reply code-challenges.test.js:85]: oops, but it looked like the code still worked?

File: code-challenges.test.js (hunk 11 above)
Comment (JLODI) [reply code-challenges.test.js:47]: Noted that test should be before function, thank you!

File: interview-questions.md (hunk 10 above)
Comment (JLODI) [reply interview-questions.md:10]: Thanks Charlean, I'll keep this in mind.

File: code-challenges.test.js (hunk 14 above)
Comment (JLODI) [reply code-challenges.test.js:126]: I think I remember using for loop at first for this cause i wasn't sure how to define index in the hof map, but I reviewed it later, and the map version of the code should be commented on line 138. Will keep in mind to use hof's more though.

**Pull request merged**